test/e2e/pull_test.go

```diff
@@ -84,6 +84,60 @@ var _ = Describe("Podman pull", func() {
 		Expect(session).Should(Exit(0))
 	})
 
+	It("podman pull and run on split imagestore", func() {
+		SkipIfRemote("podman-remote does not support setting external imagestore")
+
+		tmpDir := filepath.Join(podmanTest.TempDir, "splitstore")
+		podmanHello := "quay.io/podman/hello"
+
+		// pull to splitstore
+		session := podmanTest.Podman([]string{"pull", "--imagestore", tmpDir, podmanHello})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		// pull to graphRoot
+		session = podmanTest.Podman([]string{"pull", "quay.io/libpod/testdigest_v2s2@sha256:755f4d90b3716e2bf57060d249e2cd61c9ac089b1233465c5c2cb2d7ee550fdb"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		session = podmanTest.Podman([]string{"images", "--imagestore", tmpDir})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+		Expect(session.OutputToString()).To(ContainSubstring(podmanHello))
+		Expect(session.OutputToString()).To(ContainSubstring("quay.io/libpod/testdigest_v2s2"))
+
+		// Test deleting image in graphroot even when `--imagestore` is set
+		session = podmanTest.Podman([]string{"rmi", "--imagestore", tmpDir, "quay.io/libpod/testdigest_v2s2@sha256:755f4d90b3716e2bf57060d249e2cd61c9ac089b1233465c5c2cb2d7ee550fdb"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		session = podmanTest.Podman([]string{"images"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+		Expect(session.OutputToString()).To(Not(ContainSubstring(podmanHello)))
+
+		// Set `imagestore` in `storage.conf` and container should run.
+		configPath := filepath.Join(podmanTest.TempDir, ".config", "containers", "storage.conf")
+		os.Setenv("CONTAINERS_STORAGE_CONF", configPath)
+		defer func() {
+			os.Unsetenv("CONTAINERS_STORAGE_CONF")
+		}()
+		err := os.RemoveAll(filepath.Dir(configPath))
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
Why are you calling this? The dir should not exists at this point.

Reply:
Fixed.